ffmpeg_tools/meta.py

```diff
@@ -56,29 +54,29 @@ def get_duration(metadata, stream=0):
     return float(metadata["format"]["duration"])
 
 
-def get_video_codec(metadata):
-    for stream in metadata["streams"]:
-        if stream["codec_type"] == "video":
-            return stream["codec_name"]
-    return ""
-
+def get_codecs(
+    metadata: Dict['str', Any],
+    codec_type: str=None) -> List[Any]:
 
-def get_audio_codec(metadata):
-    for stream in metadata["streams"]:
-        if stream["codec_type"] == "audio":
-            return stream["codec_name"]
-    return ""
+    return get_attribute_from_all_streams(metadata, 'codec_name', codec_type)
 
 
 def get_format(metadata):
     return metadata["format"]["format_name"]
 
 
-def get_audio_stream(metadata):
-    for stream in metadata['streams']:
-        if stream["codec_type"] == "audio":
-            return stream
-    return None
+def get_streams(
+    metadata: Dict['str', Any],
+    codec_type: str=None) -> Dict['str', Any]:
+
+    return [
+        stream
+        for stream in metadata.get('streams', [])
+        if (
+            codec_type is None or
+            stream.get('codec_type') == codec_type
+        )
+    ]
```

Comment:
Most of this functions for quering single channel were used outside of Golem. They should remain.
But internally they should use mutlistreams functions to avoid code duplication.

Reply:
OK. I'll restore the previous functions.

Is it OK to add asserts that the file does contain only a single stream of the requested type though? These old functions only make sense if you first validate the file to make sure that the first stream is the only stream. Or do you actually rely on it sometimes ignoring the other streams?

Reply:
Done. I put the asserts in a separate commit so that they're easy to remove if you don't want them.

Reply:
I don't want asserts. I don't want to check anything before function call

Reply:
These function could have semantic: take value for n-th stream, n is 0 by default

Reply:
Assertions removed.